aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -683,13 +685,21 @@ def _to_proxy_event(self, event: Dict) -> BaseProxyEvent:
         if self._proxy_type == ProxyEventType.LambdaFunctionUrlEvent:
             logger.debug("Converting event to Lambda Function URL contract")
             return LambdaFunctionUrlEvent(event)
+        if self._proxy_type == ProxyEventType.VPCLatticeEvent:
+            logger.debug("Converting event to VPC Lattice contract")
+            return VPCLatticeEvent(event)
         logger.debug("Converting event to ALB contract")
         return ALBEvent(event)
 
     def _resolve(self) -> ResponseBuilder:
         """Resolves the response or return the not found response"""
-        method = self.current_event.http_method.upper()
-        path = self._remove_prefix(self.current_event.path)
+        if self._proxy_type == ProxyEventType.VPCLatticeEvent:
+            method = self.current_event.method.upper()
+            path = self._remove_prefix(self.current_event.raw_path)
+        else:
+            method = self.current_event.http_method.upper()
+            path = self._remove_prefix(self.current_event.path)
```

Comment:
See bellow. I believe we should treat the event the same way instead of the `if/else` condition. Bellow I have a suggestion on how to keep the interface consistence by overriding methods on the `VPCLatticeEvent`.

Reply:
Thanks for this valuable feedback, @rubenfonseca! It makes perfect sense to keep the interface consistent. :heart: